compiler/aarch64/codegen/OMRCodeGenerator.cpp

```diff
@@ -236,11 +239,27 @@ OMR::ARM64::CodeGenerator::doBinaryEncoding()
    TR_ARM64BinaryEncodingData data;
    data.estimate = 0;
    TR::Compilation *comp = self()->comp();
+   TR::Instruction *tempInstruction;
 
    self()->generateBinaryEncodingPrePrologue(data);
-   self()->getLinkage()->createPrologue(data.cursorInstruction);
 
    data.cursorInstruction = self()->getFirstInstruction();
+
+   while (data.cursorInstruction && data.cursorInstruction->getOpCodeValue() != TR::InstOpCode::proc)
+      {
+      data.estimate = data.cursorInstruction->estimateBinaryLength(data.estimate);
+      data.cursorInstruction = data.cursorInstruction->getNext();
+      }
+
+   tempInstruction = data.cursorInstruction;
+
+   if (data.recomp != NULL && !data.recomp->useSampling())
```

Comment:
This part also needs to be moved out of OMR.  `useSampling()` is defined in J9Recompilation.hpp.

Reply:
Moved the call to `useSampling()` to `TR_ARM64Recompilation::generatePrologue()`.